sources/packages/backend/apps/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -242,13 +243,18 @@ export class StudentAESTController extends BaseController {
     @Param("studentId", ParseIntPipe) studentId: number,
   ): Promise<AESTStudentProfileAPIOutDTO> {
     const [student, studentRestrictions] = await Promise.all([
-      this.studentControllerService.getStudentProfile(studentId),
+      this.studentControllerService.getStudentProfile(studentId, {
+        withSensitiveData: true,
+      }),
       this.studentRestrictionService.getStudentRestrictionsById(studentId, {
         onlyActive: true,
       }),
     ]);
 
-    return { ...student, hasRestriction: !!studentRestrictions.length };
+    return {
+      ...(student as SensitiveDataStudentProfileAPIOutDTO),
```

Comment:
The lastest PRs have been creating overloads to better explicit the return type and avoid the cast.
For instance, the method `getStudentProfile` can receive an overload and the cast can be eliminated.

Reply:
Ok. Did the overload.